.gitlab-ci.yml

```diff
@@ -162,10 +158,62 @@ check-rustdoc:
     RUSTDOCFLAGS:                  "-Dwarnings"
   script:
     - time cargo +nightly doc --workspace --all-features --verbose --no-deps
-    - sccache -s
+
+cargo-check-benches:
+  stage:                           test
+  <<:                              *docker-env
+  # this is an artificial job dependency, for pipeline optimization using GitLab's DAGs
+  needs:
+    - job:                         check-rustdoc
+      artifacts:                   false
+  script:
+    - time cargo check --all --benches
+
+#### stage:                        build
+
+build-linux-stable:
+  stage:                           build
+  <<:                              *docker-env
+  <<:                              *collect-artifacts-short
+  # this is an artificial job dependency, for pipeline optimization using GitLab's DAGs
+  needs:
+    - job:                         check-rustdoc
+      artifacts:                   false
+  script:
+    - mkdir -p ./artifacts
+    - time cargo build --release --locked --bin polkadot-parachain
+    - mv ./target/release/polkadot-parachain ./artifacts/.
 
 #### stage:                        publish
 
+build-push-image:
+  stage:                           publish
+  <<:                              *kubernetes-env
+  image:                           quay.io/buildah/stable
+  needs:
+    - job:                         build-linux-stable
+      artifacts:                   true
+  variables:
+    DOCKERFILE:                    "docker/polkadot-parachain_unsigned_injected.Dockerfile"
+    IMAGE_NAME:                    docker.io/paritypr/polkadot-parachain
```

Comment:
might make sense to name it in according to https://hub.docker.com/r/paritypr/polkadot-debug/tags

Reply:
yes

Reply:
@PierreBesson what do you think?

Reply:
he was the one who asked to rename glorious `synth-wave` into `polkadot-debug`, how do you think? :)

Reply:
Nevertheless, in the [original issue](https://github.com/paritytech/ci_cd/issues/457) the name was defined as `polkadot-parachain`